homeassistant/components/steam_online/config_flow.py

```diff
@@ -0,0 +1,222 @@
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Get the options flow for this handler."""
+        return SteamOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is None and self.entry:
+            user_input = {CONF_ACCOUNT: self.entry.data[CONF_ACCOUNT]}
+        elif user_input is not None:
+            try:
+                res = await validate_input(user_input, self.hass)
+                if res[0] is not None:
+                    name = res[0]["personaname"]
+                else:
+                    errors = {"base": "invalid_account"}
+            except (steam.api.HTTPError, steam.api.HTTPTimeoutError) as ex:
+                errors = {"base": "cannot_connect"}
+                if "403" in str(ex):
+                    errors = {"base": "invalid_auth"}
+            except Exception:  # pylint:disable=broad-except
+                errors = {"base": "unknown"}
+            if not errors:
+                entry = await self.async_set_unique_id(user_input[CONF_ACCOUNT])
+                if entry and self.source == config_entries.SOURCE_REAUTH:
+                    self.hass.config_entries.async_update_entry(entry, data=user_input)
+                    await self.hass.config_entries.async_reload(entry.entry_id)
+                    return self.async_abort(reason="reauth_successful")
+                self._abort_if_unique_id_configured()
+                if self.source == config_entries.SOURCE_IMPORT:
+                    accounts_data = {
+                        CONF_ACCOUNTS: {
+                            acc["steamid"]: {
+                                "name": acc["personaname"],
+                                "enabled": True,
+                            }
+                            for acc in res
+                        }
+                    }
+                    user_input.pop(CONF_ACCOUNTS)
+                else:
+                    accounts_data = {
+                        CONF_ACCOUNTS: {
+                            user_input[CONF_ACCOUNT]: {"name": name, "enabled": True}
+                        }
+                    }
+                return self.async_create_entry(
+                    title=name or DEFAULT_NAME,
+                    data=user_input,
+                    options=accounts_data,
+                )
+        user_input = user_input or {}
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_API_KEY, default=user_input.get(CONF_API_KEY) or ""
+                    ): str,
+                    vol.Required(
+                        CONF_ACCOUNT, default=user_input.get(CONF_ACCOUNT) or ""
```

Comment:
If this is an email address/username, it would make more sense if this is first, API key second

Reply:
It's not. It's the steamid64.